servicetalk-dns-discovery-netty/src/main/java/io/servicetalk/dns/discovery/netty/DefaultDnsClient.java

```diff
@@ -506,13 +548,25 @@ private void doQuery0() {
                     handleTerminalError0(new ClosedServiceDiscovererException(DefaultDnsClient.this +
                             " has been closed!"));
                 } else {
+                    final DnsResolutionObserver resolutionObserver;
+                    if (discoveryObserver != null) {
+                        try {
+                            resolutionObserver = discoveryObserver.newResolution(name());
+                        } catch (Throwable t) {
+                            handleTerminalError0(t);
```

Comment:
How should we react to exceptions from any method of the observer?

Currently, I fail the discovery publisher if `newDiscovery` or `newResolution` throws an exception. But log & ignore exceptions from `resolutionFailed` and `resolutionCompleted` methods of observer, because there we have additional logic that decides if we need to deliver a terminal now or not, and if we need to generate inactive events or not.

I think we can say that an observer should not affect the behavior of the discovery publisher. Therefore, we can always log & ignore exceptions from the observer methods. WDYT?

Reply:
Logging and ignoring keeps the control-flow simple. Since, we do not expect the observer methods to throw, this sounds reasonable.